File: golang/2021-06/firstBadVersion.go
package _021_06

func FirstBadVersion(n int) int {
	lo, hi := 1, n
	for lo < hi {
		mid := (lo + hi) / 2
		if !isBadVersion(mid) {
			lo = mid + 1
		}else {
			hi = mid
		}
	}
	return lo
}

File: rust/src/firstBadVersion.rs
// First Bad Version
// leetcode: https://leetcode-cn.com/problems/first-bad-version/

impl Solution {
    pub fn first_bad_version(&self, n: i32) -> i32 {
        let (mut lo, mut hi) = (1, n);
        while lo < hi {
            let mid = lo + (hi - lo) / 2;
            if !self.isBadVersion(mid) {
                lo = mid+1;
            }else {
                hi = mid
            }
        }
        lo
    }
}
